src/modules/pagemanager.js

```diff
@@ -0,0 +1,153 @@
+      task.index = newIndex;
+      newIndex += 1;
+    });
+    this.pushTasksToStorage();
+    this.loadTasksFromClass();
+  }
+
+  loadTaskItemInputListeners = () => {
+    this.changeTaskTextInputs = document.querySelectorAll('.task');
+    this.changeTaskTextInputs.forEach((input) => {
+      input.addEventListener('input', (e) => {
+        const index = e.target.parentElement.parentElement.id;
+        this.taskList[index].description = e.target.value;
+        this.pushTasksToStorage();
+      });
+    });
+    this.changeTaskTextInputs.forEach((input) => {
+      input.addEventListener('focus', (e) => {
+        const updateButtonIndex = parseInt(e.target.parentElement.parentElement.id, 10) + 1;
+        const icon = document.getElementById(`index${updateButtonIndex}`);
+        this.updateButtons.forEach((updateButton) => {
+          if (updateButton.classList.contains('trashcan')) {
+            const updateButtonIndex = parseInt(e.target.parentElement.parentElement.id, 10) + 1;
+            updateButton.classList.toggle('trashcan');
+            updateButton.innerHTML = Task.toElipses(updateButtonIndex);
+          }
+        });
+        icon.innerHTML = Task.toTrashCan(updateButtonIndex);
+        icon.classList.toggle('trashcan');
+        if (icon.getAttribute('listener') !== 'true') {
+          icon.setAttribute('listener', 'true');
+          icon.addEventListener('click', (e) => {
+            if (e.target.parentElement.id === '') {
+              e.target.parentElement.parentElement.parentElement.remove();
+              this.removeTask(parseInt(e.target.parentElement.parentElement.parentElement.id, 10));
+            } else {
+              e.target.parentElement.parentElement.remove();
+              this.removeTask(parseInt(e.target.parentElement.parentElement.id, 10));
+            }
+          });
+        }
+      });
+    });
+  }
+
+  loadUpdateButtonListeners = () => {
+    this.updateButtons = document.querySelectorAll('.update-button');
+  }
+
+  pullFromStorage = () => {
+    const tasksObj = JSON.parse(localStorage.getItem(this.storageKeyTaskList));
+    Object.values(tasksObj).forEach((task) => {
+      const index = this.taskList.length + 1;
+      const oldTask = new Task(task.description);
+      oldTask.index = index;
+      this.taskList.push(oldTask);
+    });
+  }
+
+  loadTasksFromClass = () => {
+    this.domTaskList.innerHTML = '';
+    for (let i = 0; i < this.taskList.length; i += 1) {
+      const listItem = document.createElement('li');
+      listItem.id = i;
+      listItem.innerHTML = this.taskList[i].toHTML();
+      this.domTaskList.appendChild(listItem);
+      this.checkBoxes = document.querySelectorAll('.check-box');
+    }
+    this.loadCheckBoxListeners();
+    this.loadTaskItemInputListeners();
+    this.loadUpdateButtonListeners();
+  }
+
+  addTask = () => {
+    const counter = this.taskList.length + 1;
+    const newTask = new Task(this.taskInput.value);
```

Comment:
- [ ] [OPTIONAL] you can add a check that the `this.taskInput` is empty and simply return. realistically, you wouldn't want to add an item that has no text.
```suggestion
  addTask = () => {
    if(!this.taskInput.value) return;
    const counter = this.taskList.length + 1;
    const newTask = new Task(this.taskInput.value);
```

Reply:
@MrBrN197 Good idea. I'll do that Thank you for the suggestion and the review. Cheers!